agent/agents/postgres/pgstatmonitor/stat_monitor_cache.go

```diff
@@ -151,9 +152,22 @@ func (ssc *statMonitorCache) getStatMonitorExtended(ctx context.Context, q *refo
 			newN++
 			fingerprint := c.Query
 			example := ""
+
+			var explainFingerprint string
+			var placeholdersCount uint32
+			var errParsing error
 			if !normalizedQuery {
 				example = c.Query
 				fingerprint, err = ssc.generateFingerprint(c.Query)
+				explainFingerprint, placeholdersCount, errParsing = queryparser.PostgreSQL(c.Query)
```

Comment:
fingerprint is already a normalized query, so we can just call `queryparser.PostgreSQLNormalized` with it as argument.